packages/frint/src/App.ts

```diff
@@ -0,0 +1,418 @@
+      const parentApp = app.getParentApp();
+
+      if (!parentApp) {
+        return parents;
+      }
+
+      parents.push(parentApp);
+      return findParents(parentApp, parents);
+    }
+
+    return findParents(this);
+  }
+
+  public getOption(key) {
+    return get(this.options, key);
+  }
+
+  public getName() {
+    return this.getOption('name');
+  }
+
+  public getProviders() {
+    return this.options.providers;
+  }
+
+  public getProvider(name) {
+    return find(this.options.providers, (p) => {
+      return p.name === name;
+    });
+  }
+
+  public get<T extends FrintProvider>(providerName) {
+    const value = this.container.get<T>(providerName);
+
+    if (typeof value !== 'undefined') {
+      return value;
+    }
+
+    return null;
+  }
+
+  public getApps$(regionName = null) {
+    if (!regionName) {
+      return this._apps$;
+    }
+
+    return this._apps$
+      .pipe(map$((collection) => {
+        return collection
+          .filter((w) => {
+            return w.regions.indexOf(regionName) > -1;
+          });
+      }));
+  }
+
+  public registerApp(appClass: AppClass, opts: RegisterAppOptions = {}) {
+    const options = {
+      multi: false,
+      ...opts,
+    };
+
+    if (typeof options.name !== 'undefined') {
+      Object.defineProperty(appClass, 'frintAppName', {
+        value: options.name,
+        configurable: true,
+      });
+    }
+
+    const existingIndex = findIndex(this._appsCollection, (w) => {
+      return w.name === appClass.frintAppName;
+    });
+
+    if (existingIndex !== -1) {
+      throw new Error(`App '${appClass.frintAppName}' has been already registered before.`);
+    }
+
+    this._appsCollection.push({
+      ...options,
+      name: appClass.frintAppName,
+      appClass,
+      instances: {},
+      regions: options.regions || [],
+    });
+
+    if (options.multi === false) {
+      this.instantiateApp(appClass.frintAppName);
+    }
+
+    this._apps$.next(this._appsCollection);
+  }
+
+  public hasAppInstance(name, region = null, regionKey = null) {
+    const instance = this.getAppInstance(name, region, regionKey);
+
+    if (instance && typeof instance !== 'undefined') {
+      return true;
+    }
+
+    return false;
+  }
+
+  public getAppInstance(name, region = null, regionKey = null) {
+    const index = findIndex(this._appsCollection, a => {
+      return a.name === name;
+    });
+
+    if (index === -1) {
+      return null;
+    }
+
+    const app = this._appsCollection[index];
+    const instanceKey = makeInstanceKey(region, regionKey, app.multi);
+    const instance = app.instances[instanceKey];
+
+    if (!instance || typeof instance === 'undefined') {
+      return null;
+    }
+
+    return instance;
+  }
+
+  public getAppOnceAvailable$(name, region = null, regionKey = null) {
+    const rootApp = this.getRootApp();
+
+    const w = rootApp.getAppInstance(name, region, regionKey);
+
+    if (w) {
+      return Observable.of(w);
```

Comment:
best to import the `of` module separately:

```js
import { of } from 'rxjs/observable/of';

return of(w);
```

Reply:
Done